### PR TITLE
fix: bring in node repl's top-level await

### PR DESCRIPTION
**Pull request by lrlna**
## Description
To package up our executables with built in top-level await support, we will now
process all input with [node's internal await
wrap](https://github.com/nodejs/node/blob/master/lib/internal/repl/await.j://github.com/nodejs/node/blob/master/lib/internal/repl/await.js).

This wrapper puts `async()` in front of an awaited function.

Because we are rewriting our own functions with `await`, this is done after our
internal rewrite.

### CODE REVIEW

File: packages/async-rewriter/src/async-writer-babel.ts [hunk 1]
```diff
@@ -748,4 +749,13 @@ export default class AsyncWriter {
   compile(code): string {
     return this.getTransform(code).code;
   }
+
+  /**
+   * Returns translated code.
+   * @param {string} code - string to compile.
+   */
+  process(code): string {
```
Comment (aherlihy) [inline packages/async-rewriter/src/async-writer-babel.ts:757]: Could you throw a test or two in here just to make sure it's doing what it's supposed to be doing?

Comment (lrlna) [reply packages/async-rewriter/src/async-writer-babel.ts:757]: Just did! Take a look again.

**Comment by lrlna** [comment]
@aherlihy no, it's the same as we had before. 

Since we are using the exact same implementation as node was using when we were passing `--experimental-repl-await` flag (just moved over to our side instead), parsing performance does not change.

File: packages/async-rewriter/src/async-writer-babel.ts (hunk 1 above)
Comment (aherlihy) [reply packages/async-rewriter/src/async-writer-babel.ts:757]: Looks great!

**Pull request merged**